### PR TITLE
Move deletion check to SQL command

### PR DESCRIPTION
**Pull request by rookgaard**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [X] I have read and understood the [contribution guidelines][cont] before making this PR.
- [X] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

There is no need to check in code, if player is deleted or sort them, when it's possible to do this in SQL command. 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by nekiro — DISMISSED** [review]

**Review by ranisalt — APPROVED** [review]
Good catch. Good to see you again :)

**Comment by kygov** [comment]
Are you sure? With this, you can login deleted character using packets (otclient, python library etc).

File: src/iologindata.cpp [hunk 1]
```diff
@@ -145,13 +142,13 @@ uint32_t IOLoginData::gameworldAuthentication(const std::string& accountName, co
...
 	query.str(std::string());
-	query << "SELECT `account_id`, `name`, `deletion` FROM `players` WHERE `name` = " << db.escapeString(characterName);
+	query << "SELECT `account_id`, `name` FROM `players` WHERE `name` = " << db.escapeString(characterName) << " AND `deletion` = 0";
 	result = db.storeQuery(query.str());
 	if (!result) {
 		return 0;
 	}
 
-	if (result->getNumber<uint32_t>("account_id") != accountId || result->getNumber<uint64_t>("deletion") != 0) {
+	if (result->getNumber<uint32_t>("account_id") != accountId) {
```
Comment (ranisalt) [inline src/iologindata.cpp:151]: Maybe add
```
<< "AND `account_id` = " << accountId
```
To the query and skip this branch altogether?

**Review by ranisalt — CHANGES_REQUESTED** [review]

**Comment by nekiro** [comment]
> Are you sure? With this, you can login deleted character using packets (otclient, python library etc). It makes sense in login auth, but not game world.

What? Why?

File: src/iologindata.cpp (hunk 1 above)
Comment (rookgaard) [reply src/iologindata.cpp]: Thanks, added too :)

**Comment by rookgaard** [comment]
> Are you sure? With this, you can login deleted character using packets (otclient, python library etc). It makes sense in login auth, but not game world.

Are you sure? :D

**Comment by kygov** [comment]
> > Are you sure? With this, you can login deleted character using packets (otclient, python library etc). It makes sense in login auth, but not game world.
> 
> Are you sure? :D

I dont see any check in ProtocolGame::login or ProtocolGame::onRecvFirstMessage, or I'm missing something?

**Comment by rookgaard** [comment]
@nekiro @ranisalt @Znote what are conditions when PR is actually accepted and merged?

**Review by nekiro — APPROVED** [review]
LGTM

**Comment by nekiro** [comment]
> @nekiro @ranisalt @Znote what are conditions when PR is actually accepted and merged?

We usually let it sit a bit in pull request section, so other people can throw their opinion on the changes.

**Review by EPuncker — APPROVED** [review]

**Comment by rookgaard** [comment]
> > @nekiro @ranisalt @Znote what are conditions when PR is actually accepted and merged?
> 
> We usually let it sit a bit in pull request section, so other people can throw their opinion on the changes.

Thanks! I asked, because saw some other PR's, for example, labelled by author "I didn't test it, so testing is welcome." opened and merged in less than hour https://github.com/otland/forgottenserver/pull/3382 ;)

**Comment by nekiro** [comment]
> > > @nekiro @ranisalt @Znote what are conditions when PR is actually accepted and merged?
> > 
> > 
> > We usually let it sit a bit in pull request section, so other people can throw their opinion on the changes.
> 
> Thanks! I asked, because saw some other PR's, for example, labelled by author "I didn't test it, so testing is welcome." opened and merged in less than hour #3382 ;)

We talked about it internally with @EPuncker on discord.

**Review by ranisalt — APPROVED** [review]

**Comment by ranisalt** [comment]
Thanks!

**Pull request merged**